modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb

```diff
@@ -371,20 +393,24 @@ def decision_date_not_in_past(issue, issue_index)
       add_decision_date_error "isn't in the past: #{issue.decision_date_string.inspect}", issue_index
     end
 
+    # rubocop:disable Layout/LineLength
     def add_decision_date_error(string, issue_index)
-      add_error "included[#{issue_index}].attributes.decisionDate #{string}"
-    end
-
-    def add_error(message)
-      errors.add(:base, message)
+      add_error("included[#{issue_index}].attributes.decisionDate #{string}", title: 'Invalid field value',
+                                                                              missing_fields: ['decision date'],
+                                                                              pointer: '/data/attributes/included/items/attributes/properties',
+                                                                              code: 144)
     end
+    # rubocop:enable Layout/LineLength
 
     def header_field_as_string(key)
       auth_headers&.dig(key).to_s.strip
     end
 
     def validate_api_version_presence
-      add_error('Appeal must include the api_version attribute') if api_version.blank?
+      if api_version.blank?
+        add_error('Appeal must include the api_version attribute', missing_fields: ['api_version'],
```

Comment:
FYI this particular validation is more for us to make sure our controller has set the API version correctly - it is not something the consumer can supply themselves and so they should never encounter it. 

Still, better safe than sorry!  I don't really have a good alternative for this right now so I won't block it - but maybe it's something to discuss in our huddle?

Reply:
@drew, I believe I added this after a convo we had about the confusion surrounding when and where the api version is set. Let's definitely come together to discuss.

Reply:
@ElleeB Ah, that's right!

@stiehlrod I'm fine with leaving this message here, or we can pull it out and add a comment to the method detailing why it's special. Up to you!